use a HANDSHAKE_DONE frame to drive the handshake to confirmation

marten-seemann opened the pull request.

bnc-google committed:
s/up to/up to and including/ for clarity.

This was suggested by dtikhonov at
https://github.com/quicwg/base-drafts/pull/2942/files/11edccf6ed3697ba785578712b54a42051bb090d#r337177688.

bnc-google committed:
Make language of representations consistent.

This was inspired by the comment by afrind at
https://github.com/quicwg/base-drafts/pull/2942#discussion_r335222617.

For each representation, uniformly
* use the word "representation" in the first sentence;
* use the word "identifies" if it's a verbatim entry; "encodes" otherwise;
* identify relative versus post-base indexing in the first sentence;
* move drawing right after first sentence;
* use consistent language describing binary format.

marten-seemann committed:
use a HANDSHAKE_DONE frame to drive the handshake to confirmation

kazuho commented on draft-ietf-quic-tls.md:
```diff
@@ -385,13 +385,9 @@ perspective of the endpoint in question.
-in any received 1-RTT ACK frame: once the latter is higher than or equal to the
-former, the handshake is confirmed.
+In this document, the TLS handshake is considered confirmed at the server when
+the handshake completes. At the client, the handshake is consider confirmed when
```
s/is consider/is considered/

marten-seemann replied:
Thanks. Fixed.

marten-seemann committed:
more concise timing requirement

marten-seemann committed:
move retransmission rule for HANDSHAKE_DONE frames

marten-seemann committed:
prohibit clients from sending HANDSHAKE_DONE

amirlivneh committed:
Fix typos in QPACK draft

kazuho committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

kazuho committed:
wordwrap

ianswett committed:
Don't arm PTO for ApplicationData until complete

ianswett committed:
Initial and Handshake have no max_ack_delay

ianswett committed:
MUST limit bursts to the initial congestion window (#3160)

* MUST limit bursts to the initial congestion window

From discussion on #3106 
https://github.com/quicwg/base-drafts/pull/3106#discussion_r338921935

* Update draft-ietf-quic-recovery.md

* Update draft-ietf-quic-recovery.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Martin Duke's suggestions

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
kMaxDatagramSize -> max_datagram_size

Clarifies that this is the maximum datagram size currently in use on the connection.

Fixes #3041

martinthomson committed:
Merge pull request #3108 from DavidSchinazi/no_tls_pres

Replace use of the TLS presentation language with QUIC encoding

MikeBishop committed:
Feedback

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Cite 14.1 of Transport

martinthomson committed:
Merge pull request #3138 from quicwg/if-0rtt

Applications need to disable or selectively reject 0-RTT

martinthomson committed:
Fix handling of Retry in recovery (#3148)

* Fix handling of Retry in recovery

This section was well out of date.  Also, this covers what needs to be
retained across this transition better.  Basically, the only thing that
needs to go is the congestion and loss recovery state.  The rest can
stay; in fact, the crypto handshake has to stay.

Closes #2058.

* cannot/MUST NOT

potato/tomato

Co-Authored-By: ianswett <ianswett@users.noreply.github.com>

MikeBishop committed:
Talk about concurrency in H3 vs. H2 (#3114)

* Talk about concurrency

* Talk about push separately

* Precision, thanks to @kazuho

* Update draft-ietf-quic-http.md

Co-Authored-By: Lucas Pardue <lucaspardue.24.7@gmail.com>

ianswett committed:
Update draft-ietf-quic-recovery.md

martinthomson committed:
Merge pull request #3149 from quicwg/transport/name_connections

Explicit linkages for ODCID/OCID TP

martinthomson committed:
Merge pull request #3165 from quicwg/reset-after-fin

Don't require RESET_STREAM on terminated streams

ianswett committed:
Update draft-ietf-quic-recovery.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

ianswett committed:
Jana's suggestion

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Update draft-ietf-quic-recovery.md

janaiyengar committed:
Merge pull request #3057 from quicwg/ianswett-must-retransmit

PTO MUST send new data or retransmit data if possible

kazuho committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

martinthomson committed:
an app that wishes to do so can

kazuho committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Mike Bishop <mbishop@evequefou.be>

martinthomson committed:
Allow provisional registrations for all registries

This overhauls the system for IANA registries to use a unified scheme
that allows provisional registration.  This seems pretty complicated in
terms of nailing down all the corner cases, but it is fairly simple in
practice:

* Provisional registration is easy, it only requires that the requester
provide contact information and (ideally) the codepoint they want.
They could provide more, but at their discretion.

* Permanent registration requires that you fill out the fields.

Experts are required for both, but only really to prevent spam.  We have
a few more requirements for permanent registrations of frame types, but
that is more for the purposes of advice, because sending an new frame
leads to protocol errors if you haven't agreed to that in the past.

I've sketched out a strawman for a codepoint reclamation process.  That
advises that experts try to find some use of the codepoint.  There is a
date on registrations that can guide the selection of codepoints to
reclaim, but these can be updated any time by anyone who wants to keep
the registration "live".

Closes #3109. Closes #3020.

kazuho committed:
application does not send a frame

ianswett committed:
Update draft-ietf-quic-recovery.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

ianswett committed:
Update draft-ietf-quic-recovery.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

ianswett committed:
Update draft-ietf-quic-recovery.md

martinthomson committed:
enstricted

kazuho committed:
Update draft-ietf-quic-tls.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

ianswett committed:
Retry MUST NOT -> cannot be treated as an acknowledgement (#3171)

* Retry MUST NOT -> cannot be treated as an acknowledgement

I didn't notice this last commit in @martinthomson PR, but I think the older text is more correct.  I added a bit more detail about why it cannot be treated as an acknowledgement.

* Martin's reason

* Update draft-ietf-quic-recovery.md

* Update draft-ietf-quic-recovery.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

martinthomson committed:
Narrow scope of draft caching

martinthomson committed:
Merge pull request #3098 from kazuho/kazuho/connection-close-not-ack-eliciting

CONNECTION_CLOSE is non-ack-eliciting

martinthomson committed:
Merge pull request #2993 from quicwg/ct-sr

Stateless reset comparisons (constant time/any order/datagram)

martinthomson committed:
Merge pull request #3096 from kazuho/kazuho/retire-prior-to-must

MUST retire Connection IDs becoming stale

martinthomson committed:
Merge pull request #2870 from quicwg/retry-rekey

Initial secrets change after Retry

martinthomson committed:
Merge pull request #3045 from DavidSchinazi/multi_packet_client_hello

Allow ClientHello to span multiple QUIC packets

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Update draft-ietf-quic-recovery.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

ianswett committed:
loss_times -> loss_time

martinthomson committed:
Move some text around

martinthomson committed:
Transport and TLS changelog for -24

martinthomson committed:
Add workaround for caching bug

martinthomson committed:
janaiyengar's suggestions

kazuho committed:
"the" @MikeBishop

kazuho committed:
add unless, based on the discussion starting from https://github.com/quicwg/base-drafts/issues/3155#issuecomment-548264247

janaiyengar committed:
Merge pull request #3167 from quicwg/ianswett-max-datagram-size

kMaxDatagramSize -> max_datagram_size

ianswett committed:
Merge branch 'master' into ianswett-pto-per-pn-space

ianswett committed:
1 loss_times -> loss_time

Commit:
Further specify frame Length field (#3177)

* Further specify frame Length field

Probably a bit overkill, but since it wasn't explicitly stated + I thought we had replaced "octets" with bytes everywhere else.

* remove brackets

Co-Authored-By: ianswett <ianswett@users.noreply.github.com>

MikeBishop committed:
HTTP changelog

MikeBishop committed:
Nothing non-editorial for QPACK

ianswett committed:
Kazuho's suggestion

About being able to unprotect the Initial packet

ianswett committed:
Update draft-ietf-quic-transport.md

ianswett committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

ianswett committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

ianswett committed:
MAY close the connection without creating state

ianswett committed:
Jana's suggestion

ianswett committed:
Update draft-ietf-quic-transport.md

ianswett committed:
Update draft-ietf-quic-transport.md

ianswett committed:
Reword an awkward sentence

ianswett committed:
Recovery changelog for -24

Includes #3066, which I plan to merge Monday morning.

martinthomson committed:
Some small adjustments

martinthomson committed:
Merge branch 'master' into rework-key-update-2

martinthomson committed:
Split first from subsequent again

martinthomson committed:
Jana's suggestions

martinthomson committed:
Merge pull request #3181 from quicwg/http/changelog-24

HTTP & QPACK changelogs for -24

martinthomson committed:
Merge pull request #3180 from quicwg/tt-changelog-24

Transport and TLS changelog for -24

martinthomson committed:
Merge pull request #3050 from quicwg/rework-key-update-2

Rewrite key update section

martinthomson committed:
Merge pull request #3185 from quicwg/ianswett-recovery-changelog-24

Recovery changelog for -24

marten-seemann committed:
add an error for overflow of a CRYPTO stream

marten-seemann committed:
add an error for overflow of a MAX_STREAMS frame

marten-seemann committed:
add an error for overflow of a STREAMS_BLOCKED frame

martinthomson committed:
Merge pull request #3042 from marten-seemann/frame-encoding-errors

Use the FRAME_ENCODING_ERROR for errors in frame encoding

martinthomson committed:
Update changelog for #3042

marten-seemann committed:
allow CRYPTO_BUFFER_EXCEEDED when overflowing a CRYPTO frame

ad-l committed:
Fix for #3187

martinthomson committed:
Add ;

martinthomson committed:
Remove #3066 from recovery changelog

martinthomson committed:
Merge pull request #3190 from quicwg/see-downgrade

Add a few semicolons

ad-l committed:
Integrate Martin Thomson's comments

Commit:
Add ClientHello changes to transport changelog since -23

ianswett committed:
Jana's comments

martinthomson committed:
Merge pull request #3188 from ad-l/pn_decoding_overflow

Correct overflow checks on packet number decoding algorithm

janaiyengar committed:
Merge pull request #3066 from quicwg/ianswett-pto-per-pn-space

Change PTO to be per packet number space

MikeBishop committed:
Don't call two different things "blocking" (#3172)

* Evictable, not blocking

* Focus on insertion, not eviction

bencebeky committed:
Fix bug in Insert Count Increment description. (#3132)

* Fix bug in Insert Count Increment description.

Fixes issue 3082.

This PR fixes the race condition and clarifies that both endpoints must
keep track of the Known Received Count.

* Revert some changes.

* Remove unnecessarily inserted blank line.

martinthomson committed:
Merge pull request #3191 from dtikhonov/201911051348-allow-CH-span-to-changelog

Add ClientHello changes to transport changelog since -23

martinthomson committed:
Merge pull request #3036 from marten-seemann/stronger-retire-error-check

require peers to check if RETIRE_CONNECTION_ID sequence number is valid

martinthomson committed:
Certainly

Fixes #2430.

martinthomson committed:
Editorial fixes from @MikeBishop

Co-Authored-By: Mike Bishop <mbishop@evequefou.be>

janaiyengar committed:
Merge pull request #3192 from quicwg/probably

Certainly

martinthomson committed:
Add default contact details

martinthomson committed:
Add HTTP/3 registration updates

MikeBishop committed:
Semicolons before "see {section}"

marten-seemann committed:
forbid exceeding the peer's active_connection_id_limit

marten-seemann committed:
add an error code for exceeding the connection ID limit

marten-seemann committed:
handle reordered NEW_CONNECTION_ID frames

marten-seemann committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

Commit:
Add Retry CID change requirement to transport changelog

marten-seemann committed:
Ian's and Eric's suggestion

marten-seemann committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

marten-seemann committed:
set a default and minimum value for active_connection_id_limit

ekinnear committed:
Consolidate Retire Prior To instructions in section, clean up frame description

marten-seemann committed:
mandate the processing order of NEW_CONNECTION_ID frame fields

MikeBishop committed:
Minor formatting

MikeBishop committed:
Recommend H3_NO_ERROR for early responses

martinduke committed:
PING in all contexts

#3035 updated quic-tls but not quic-transport

MikeBishop committed:
Clarify what the error code is for

martinduke committed:
Wrapped line

Wrapped line

martinthomson committed:
Merge pull request #3209 from martinduke/patch-2

PING in all contexts

erickinnear committed:
Martin's suggestion

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

ekinnear committed:
Reword to match Martin's suggestion

MikeBishop committed:
Title-Case Titles (#3207)

bencebeky committed:
[HTTP/3] [editorial] s/each request and response/each request-response pair/ (#3222)

Lest some readers get the impression that each request consumes a single
QUIC stream and each response consumes a single QUIC stream.

martinthomson committed:
Merge pull request #3204 from erickinnear/ek/cleanup_retire_prior_to

Resolve conflicting advice on how to handle Retire Prior To

martinthomson committed:
Merge pull request #3203 from dtikhonov/201911071345-add-2837-and-3147-to-transport-changelog

Add Retry CID change requirement to transport changelog

ianswett committed:
RTTVar is not really a variance

It's a mean deviation.  This has confused at least one reader.

ianswett committed:
Martin's suggestion

ianswett committed:
Clarify loss epoch (#3213)

* Clarify loss epoch

Fixes #3210

* Another try

ianswett committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

ianswett committed:
Update draft-ietf-quic-transport.md

martinthomson committed:
No reading of 1-RTT before the handshake is complete

We really mean it.

The effect of #3159 has been to shame some implementations into the
right behaviour, but on reviewing this section, it really was
unambiguous.  This tweaks the text slightly and adds a cautionary note,
but it is otherwise unchanged other than a few editorial tweaks.

Closes #3159.

martinthomson committed:
A day in the life

...and now we know how many holes it takes to fill the Albert Hall.

This section has existed for a long time as "TBD".  No longer.  It also
takes the opportunity to explain a little bit about what 0-RTT is for.

Closes #2856.
Closes #2843.

martinthomson committed:
Clearer text for application errors

This cribs Lucas' proposed tweak and makes a few other tiny adjustments.

Closes #2850.

martinthomson committed:
Summarize where frames can be used

It's still slightly odd where we have this setup between TLS and
transport, but it seems like replicating the information might work.

Closes #2721.

martinthomson committed:
Simplify

martinthomson committed:
grammer iz had

martinthomson committed:
Restore other state after a spoofed migration

This is really just an observation, because there is no real way we can
insist on people doing this sort of thing, but being aware of the
performance implication of spoofed migrations might motivate solutions.

Closes #2311.

martinthomson committed:
Use no_application_protocol more

The old text used the non-specific "abort" here, but we can use the
right error code.

Closes #2500.

martinthomson committed:
Yes, I meant to remove that 'if'.  Oops.

martinthomson committed:
Bad CONNECTION_CLOSE isn't special

If we get one, rely on the back-off that we require endpoints to
implement when they enter the closing state.  No need for any additional
special processing rules.

Closes #2475.
Closes #3230.

bencebeky committed:
[HTTP/3] [editorial] How many parts is the HTTP/3 specification is split into exactly? (#3219)

I count eight including the additional resources or six not including
them.  But what I'm most concerned about is that Sections 1 and 2 are
also part of the HTTP/3 specification.

martinthomson committed:
Mikkel makes this better

Co-Authored-By: MikkelFJ <mikkelfj@gmail.com>

martinthomson committed:
only the application

Co-Authored-By: Mike Bishop <mbishop@evequefou.be>

martinthomson committed:
Reorder, tweak

martinthomson committed:
termination clause

Co-Authored-By: ianswett <ianswett@users.noreply.github.com>

martinthomson committed:
Paraphrase Mike's suggestion

bencebeky committed:
[HTTP/3] [editorial] Remove mention of redefining setting format. (#3220)

* [HTTP/3] [editorial] Remove mention of redefining setting format.

Each SETTINGS parameter consists of an identifier and a value, each
encoded as QUIC variable-length integer.  The sentence being removed is
a remnant from the time when setting values were encoded as
length-prefixed binary blobs.

* Clarify how settings ported from HTTP/2 might redefine value.

* reword parenthetical

ianswett committed:
Recovery editorial update

Update based on my review of the editor's copy.

ianswett committed:
whitespace

ianswett committed:
Update draft-ietf-quic-recovery.md

martinthomson committed:
Update vectors for draft-25

ianswett committed:
App or flow control limited (#3238)

* App or flow control limited

Fixes #3236

* spelling

ianswett committed:
Fix Gorry's editorial comments (#3237)

This attempts to fix Gorry's editorial comments from the QUIC mailing list.

ianswett committed:
MUST NOT use a reordering theshold less than 3

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Reference 3168 for ECN

And note that 8311 is an update.

marten-seemann committed:
apply David's suggestions, with minor modifications

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
ECN editorial

From Gorry's comments

marten-seemann committed:
send HANDSHAKE_DONE immediately

marten-seemann committed:
include the handshake and SPA CIDs in the connection ID limit

marten-seemann committed:
rephrase when to supply new CIDs

martinthomson committed:
--layer

martinthomson committed:
Use host less as appropriate

martinthomson committed:
Define "address"

Closes 3233.

janaiyengar committed:
Fix incorrect expression

marten-seemann committed:
Merge remote-tracking branch 'upstream/master' into enforce-active-connection-id-limit

janaiyengar committed:
Set, not seed

ianswett committed:
Jana's suggestion

ianswett committed:
Reference 3168 for ECN (#3241)

* Reference 3168 for ECN

And note that 8311 is an update.

* Update draft-ietf-quic-recovery.md

ianswett committed:
Merge pull request #3251 from quicwg/jri/not-seed

Clarify that initial RTT is set, not seeded into something

ianswett committed:
Merge pull request #3250 from quicwg/jri/time-threshold

Fix incorrect time-threshold expression

kazuho committed:
endpoints never requests a CID

martinthomson committed:
Avoid using table- prefixes for anchors

xml2rfc v3 format prohibits the use of these, which will cause problems.
This makes the anchors longer, so feel free to go with something else.

martinthomson committed:
Merge pull request #3252 from kazuho/kazuho/endpoint-never-request-cid

endpoints never requests a CID

ianswett committed:
Merge pull request #3242 from quicwg/ianswett-ecn-editorial

ECN editorial

marten-seemann committed:
no active_connection_id_limit when using zero-length CIDs

marten-seemann committed:
apply mt's suggestion

marten-seemann committed:
Update draft-ietf-quic-transport.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

martinthomson committed:
Merge pull request #3253 from quicwg/table-tag

Avoid using table- prefixes for anchors

martinthomson committed:
Remove some cruft from packet size section

This section had a bunch of text that wasn't updated as the result of
other changes elsewhere.  Though there was a suggestion that this could
reference other sections, but as this is in some ways the authoritative
section on the subject, I decided to keep the requirements.  That's
duplicative, but I couldn't see a simple way to deal with that.

Closes #3255.

martinthomson committed:
Key generation for updates MAY be deferred

As concluded in Singapore.

Closes #2792.

martinthomson committed:
Keep context

marten-seemann committed:
apply Jana's suggestion

Lekensteyn committed:
TLS: fix references to registries

martinthomson committed:
Allow ranges

ianswett committed:
Merge pull request #3234 from quicwg/ianswett-recovery-editorial

Recovery editorial update

martinthomson committed:
Merge pull request #3267 from Lekensteyn/tls-iana-fixes

TLS: fix references to registries

janaiyengar committed:
Merge pull request #3249 from quicwg/sub-host

Use host less, define address more

mnot committed:
Update CONTRIBUTING.md

martinthomson committed:
iyengareditorialization

martinthomson committed:
Merge pull request #3262 from quicwg/remove-first-initial

Remove some cruft from packet size section

martinthomson committed:
A zero-length Retry Token is invalid

The Retry packet can be discarded.

Closes #3271.

martinthomson committed:
Merge pull request #3157 from kazuho/kazuho/connection-close-must-backoff

Backoff on CONNECTION_CLOSE

martinthomson committed:
Merge pull request #3231 from quicwg/bad-close-ok

Bad CONNECTION_CLOSE isn't special

martinthomson committed:
Merge pull request #3156 from kazuho/kazuho/token-scope

clarify the scope of a NEW_TOKEN token

martinthomson committed:
Merge pull request #3164 from kazuho/kazuho/application-close-only-in-1rtt

Forbid use of application-level close in Initial, Handshake packets

martinthomson committed:
Merge pull request #3128 from quicwg/must-distinguish-tokens

Mandatory distinction between token types

martinthomson committed:
Merge branch 'master' into handle-reordered-ncid-frames

martinthomson committed:
Merge pull request #3202 from marten-seemann/handle-reordered-ncid-frames

handle reordered NEW_CONNECTION_ID frames

martinthomson committed:
Fix offset of Packet Number field in example

Closes #3279.

martinthomson committed:
Merge pull request #3225 from quicwg/life

A day in the life

martinthomson committed:
Merge pull request #3229 from quicwg/restore-path-state

Restore other state after a spoofed migration

martinthomson committed:
Merge pull request #3228 from quicwg/abort

Use no_application_protocol more

martinthomson committed:
Merge pull request #3226 from quicwg/resetting

Clearer text for application errors

martinthomson committed:
Merge pull request #3139 from erickinnear/ek/reset_cc

Packets on one path must not adjust values for a different path

martinthomson committed:
Merge pull request #3107 from quicwg/ianswett-invalid-token

Immediately close with INVALID_TOKEN

martinthomson committed:
Merge pull request #3277 from quicwg/retry-token-zero

A zero-length Retry Token is invalid

martinthomson committed:
Merge pull request #3196 from quicwg/semicolons

Semicolons before "see {section}"

martinthomson committed:
NEW_TOKEN contains globally unique values

This is already implied by the existing text, but it isn't said
directly.  That is, we say that the values can't be linkable, which
is a stronger requirement.  Saying that you can't send the same value on
two connections is much clearer.

Closes #3179.

martinthomson committed:
Merge pull request #3284 from quicwg/vector-25

Update vectors for draft-25

janaiyengar committed:
Rephrase sentence

martinthomson committed:
Merge pull request #3285 from quicwg/jri/edit

Reword sentence about backoff of CONNECTION_CLOSE

martinthomson committed:
Paraphrase suggestions

martinthomson committed:
Merge pull request #3263 from quicwg/may-defer-keygen

Key generation for updates MAY be deferred

martinthomson committed:
Merge branch 'master' into provisional

martinthomson committed:
Merge pull request #3208 from quicwg/http/early_response

Recommend H3_NO_ERROR for early responses

martinthomson committed:
Merge pull request #3224 from quicwg/really-really-dont-1rtt-early

No reading of 1-RTT before the handshake is complete

martinthomson committed:
Merge pull request #3201 from marten-seemann/enforce-active-connection-id-limit

enforce the active_connection_id_limit

martinthomson committed:
Merge pull request #3240 from quicwg/ianswett-must-not-3

SHOULD NOT use a reordering threshold less than 3

martinthomson committed:
Merge pull request #3186 from marten-seemann/crypto-frame-overflow

allow CRYPTO_BUFFER_EXCEEDED when overflowing a CRYPTO frame

martinthomson committed:
Merge pull request #3170 from quicwg/provisional

Allow provisional registrations for all registries

martinthomson committed:
CONNECTION_CLOSE in Handshake too

martinthomson committed:
Reword, thanks @janaiyengar

martinthomson committed:
...and 1-RTT too

martinthomson committed:
Merge pull request #3293 from quicwg/close-in-handshake

CONNECTION_CLOSE in Handshake too

martinthomson committed:
Remove duplicated CONNECTION_CLOSE rules

MikeBishop committed:
Recommend replacing, not changing, protocol components (#3206)

* Recommend replacing, not changing

* Remove extra word

* Update draft-ietf-quic-http.md

Co-Authored-By: Martin Thomson <mt@lowentropy.net>

ianswett committed:
Update draft-ietf-quic-recovery.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

ianswett committed:
Update draft-ietf-quic-recovery.md

Co-Authored-By: Jana Iyengar <jri.ietf@gmail.com>

janaiyengar committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Update draft-ietf-quic-recovery.md

ianswett committed:
Reduce repetition

ianswett committed:
Merge pull request #3223 from quicwg/ianswett-mean-deviation

RTTVar is not really a variance

janaiyengar committed:
Update draft-ietf-quic-transport.md

janaiyengar committed:
Merge pull request #3281 from quicwg/dedupe-token

NEW_TOKEN contains globally unique values

martinthomson committed:
Move the retained sentence higher instead

janaiyengar committed:
Merge pull request #3295 from quicwg/remove-duped-close-text

Remove duplicated CONNECTION_CLOSE rules

martinthomson committed:
Move alternative confirmation method

janaiyengar committed:
editorial suggestion

martinthomson committed:
Merge pull request #3299 from quicwg/janaiyengar-patch-1

 editorial suggestion

martinthomson committed:
Merge pull request #3227 from quicwg/where-art-thou-frame

Summarize where frames can be used

martinthomson committed:
Add HANDSHAKE_DONE to the table; fix anchor

martinthomson committed:
Remove cruft

marten-seemann committed:
apply @ianswett's suggestion

Co-Authored-By: ianswett <ianswett@users.noreply.github.com>

marten-seemann committed:
apply @MikeBishop's suggestions

Co-Authored-By: Mike Bishop <mbishop@evequefou.be>
